s2spy/calendar_shifter.py

```diff
@@ -0,0 +1,169 @@
+import copy
+from typing import Dict
+from typing import List
+from typing import Union
+import xarray as xr
+import s2spy.time
+
+
+def _gap_shift(
+    interval: s2spy.time.Interval, shift: Union[str, Dict[str, int]]
+) -> Dict[str, int]:
+    """
+    Shift a gap from a calendar interval by an input of a pandas-like
+    frequency string (e.g. "10d", "2W", or "3M"), or a pandas.DateOffset
+    compatible dictionary such as {days=10}, {weeks=2}, or {months=1, weeks=2}..
+    Args:
+        gap: a pandas DateOffset from a calendar instance.
+        shift: the shift for the gap
+    Returns:
+        dict with gap offset by shift
```

Comment:
```suggestion
    Shift a calendar interval's gap property by the given amount.

    Args:
        gap: the pandas DateOffset from a calendar interval's `gap` property.
        shift: the shift for the gap, in the form of a pandas-like frequency
            string (e.g. "10d", "2W", or "3M"), or a pandas.DateOffset compatible
            dictionary such as {days=10}, {weeks=2}, or {months=1, weeks=2}.

    Returns:
        A Pandas DateOffset compatible dictionary, with the gap offset by shift
```
Google Style Python Docstrings need to have the empty lines, and a single-line descriptor as well. The main body is optional, and I moved it to the argument's description where it fits better.